autopush/endpoint.py

```diff
@@ -411,14 +415,25 @@ def delete(self, token):
         self.version = self._client_info['version'] = token
         d = deferToThread(self.ap_settings.fernet.decrypt,
                           self.version.encode('utf8'))
-        d.addCallback(self._token_valid, self._delete_message)
+        d.addCallback(self._delete_message, token)
         d.addErrback(self._token_err)
         d.addErrback(self._response_err)
         return d
 
-    def _delete_message(self, kind, uaid, chid):
-        d = deferToThread(self.ap_settings.message.delete_message, uaid,
-                          chid, self.version)
+    def _delete_message(self, result, token):
+        key_info = WebPushNotification.parse_decrypted_message_id(result)
```

Comment:
I mainly wanted to have a single method that could invert what the generate_message_id method does that encrypts it. It doesn't make a complete proper WebPushNotification as it leaves out bits that aren't needed for deleting it.